src/v/net/server_probe.h

```diff
@@ -50,6 +50,8 @@ class server_probe {
 
     void timeout_waiting_rate_limit() { ++_declined_new_connections; }
 
+    void waiting_for_conection_rate() { ++_connections_wait_rate; }
```

Comment:
This won't appear in output unless you refer to it from `setup_metrics`.  Once that's added, I suggest also adding a check on the metric from the ducktape test (can just check that it is >0 after the test that hits some rate limiting)